examples/hybrid_quantum_algorithms/Quantum_Monte_Carlo_Chemistry/afqmc/quantum_afqmc_pennylane.py

```diff
@@ -0,0 +1,704 @@
+    Args:
+        Q (np.ndarray): orthonormalized walker state
+        V_T (function): quantum trial state
+    """
+    circuit_first_half(Q)
+    circuit_second_half_imag(Q, V_T)
+
+
+def amplitude_estimate(Q, V_T, dev):
+    """This function computes the amplitude between walker state and quantum trial state.
+    Args:
+        Q (np.ndarray): orthonormalized walker state
+        V_T (function): quantum trial state
+        dev (qml.device): qml.device('lightning.qubit', wires=wires) for simulator;
+                          qml.device('braket.aws.qubit', device_arn=device_arn, wires=wires, shots=shots)
+                          for quantum device;
+    Returns:
+        amplitude: numpy.complex128
+    """
+    num_qubits, num_particles = Q.shape
+
+    @qml.qnode(dev, interface=None, diff_method=None)
+    def compute_real(Q, V_T):
+        amplitude_real(Q, V_T)
+        return qml.probs(range(num_qubits))
+
+    probs_values = compute_real(Q, V_T)
+    real = probs_values[0] - probs_values[int(2**num_qubits / 2)]
+
+    @qml.qnode(dev, interface=None, diff_method=None)
+    def compute_imag(Q, V_T):
+        amplitude_imag(Q, V_T)
+        return qml.probs(range(num_qubits))
+
+    probs_values = compute_imag(Q, V_T)
+    imag = probs_values[0] - probs_values[int(2**num_qubits / 2)]
+
+    return real + 1.0j * imag
+
+
+def U_circuit(U):
+    """Construct circuit to perform unitary transformation U."""
+
+    decomposition, diagonal = givens_decomposition_square(U)
+    circuit_description = list(reversed(decomposition))
+
+    for i in range(len(diagonal)):
+        qml.RZ(np.angle(diagonal[i]), i)
+
+    if circuit_description != []:
+        prepare_slater_circuit(circuit_description)
+
+
+def pauli_real(Q, V_T, U, pauli):
+    """Construct the the vacuum reference circuit for measuring expectation value of a pauli real part
+    Args:
+        Q : orthonormalized walker state
+        V_T: quantum trial state
+        U: unitary transformation to change the Pauli into Z basis
+        pauli: list that stores the position of the Z gate, e.g., [0,1] represents 'ZZII'.
+    """
+    circuit_first_half(Q)
+
+    U_circuit(U)
+    for i in pauli:
+        qml.PauliZ(wires=i)
+
+    qml.adjoint(U_circuit)(U)
+    circuit_second_half_real(Q, V_T)
+
+
+def pauli_imag(Q, V_T, U, pauli):
+    """Construct the the vacuum reference circuit for measuring expectation value of a pauli imaginary part
+    Args:
+        Q: orthonormalized walker state
+        V_T: quantum trial state
+        U: unitary transformation to change the Pauli into Z basis
+        pauli: list that stores the position of the Z gate, e.g., [0,1] represents 'ZZII'.
+    """
+    circuit_first_half(Q)
+
+    U_circuit(U)
+    for i in pauli:
+        qml.PauliZ(wires=i)
+
+    qml.adjoint(U_circuit)(U)
+    circuit_second_half_imag(Q, V_T)
+
+
+def pauli_estimate(Q, V_T, U, pauli: list, dev):
```

Comment:
inconsistent type annotations throughout, variables/parameters should generally be snake case. Not sure that the formatting is a priority for helper files though

Reply:
@ajberdy In scientifically, it is a convention to use upper and lower cases to differentiate different object in equations. In this example, allowing underscores and upper case adds readability to code because the correspondence to the equations is more clear. I suggest that we can keep the case in variable name.